modules/ip4s/src/main/scala/pureconfig/module/ip4s/package.scala

```diff
@@ -1,31 +1,58 @@
 package pureconfig.module
 
-import com.comcast.ip4s.{Hostname, Port}
+import com.comcast.ip4s._
 
 import pureconfig.error.CannotConvert
 import pureconfig.{ConfigReader, ConfigWriter}
 
 package object ip4s {
 
+  private def mkConfigReader[From, To](f: From => Option[To])(
+      cname: String,
+      uname: String
+  )(implicit From: ConfigReader[From]): ConfigReader[To] = From.emap { from =>
+    f(from).toRight(CannotConvert(from.toString, cname, s"Invalid $uname"))
```

Comment:
Instead of manually writing the class name of `To` every time, can you please make the compiler do that for you by requiring an implicit `ClassTag[To]`?

With a way to get the class name implicitly, I also question the value of asking for a human-readable name just to put in error messages. I'd much rather have an API that simply takes the reader function and handles the rest automatically.

Reply:
agree. fixed.